Library/Homebrew/dev-cmd/test.rb

```diff
@@ -84,12 +85,19 @@ def test
       rescue ::Test::Unit::AssertionFailedError => e
         ofail "#{f.full_name}: failed"
         puts e.message
-      rescue Exception => e # rubocop:disable Lint/RescueException
+        failed_formulae << f
+      rescue StandardError => e
         ofail "#{f.full_name}: failed"
         puts e, e.backtrace
+        failed_formulae << f
       ensure
         ENV.replace(env)
       end
     end
+    return false if failed_formulae.empty?
+    ofail <<-EOF.undent
+      The following formulae failed to complete the tests successfully:
```

Comment:
This should not be output if a single formula was passed (which is the typical case which again makes me wonder if this is a useful change, I'm afraid).